### PR TITLE
[PRIMA] add recipe

### PR DESCRIPTION
**Pull request by amontoison**
@zaikunzhang

### CODE REVIEW

File: P/PRIMA/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libprimac", :libprimac),
+    LibraryProduct("libprimaf", :libprimaf)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
```
Comment (ararslan) [inline P/PRIMA/build_tarballs.jl:34]: As suggested by the build log:
```
┌ Warning: To ensure that the correct version of libgfortran is found at runtime, add the following entry to the list of dependencies of this builder
│
│     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
└ @ BinaryBuilder.Auditor /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilder/wIfP6/src/auditor/compiler_abi.jl:4
```

```suggestion
dependencies = Dependency[
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
]
```

**Comment by ararslan** [comment]
The build failure on Windows seems to be because it's building static libraries (`.a`) but not dynamic (`.dll`). Is there an additional build flag that needs to be passed?

The ARM failures on musl Linux are strange: it's complaining that it can't find an `ieee_arithmetic` intrinsic module. Apparently that was new in GFortran 5, and yet the `libgfortran4` builds succeeded for all other platforms but `libgfortran5` still fails on ARM musl Linux. Even more confusingly, i686 and x86_64 musl Linux builds succeeded.

**Comment by giordano** [comment]
https://github.com/JuliaPackaging/Yggdrasil/blob/490f462fd968fa4ce97e9e9f288888c5dfa7f613/O/OpenFAST/build_tarballs.jl#L45-L46

**Comment by zaikunzhang** [comment]
> The ARM failures on musl Linux are strange: it's complaining that it can't find an `ieee_arithmetic` intrinsic module. Apparently that was new in GFortran 5, and yet the `libgfortran4` builds succeeded for all other platforms but `libgfortran5` still fails on ARM musl Linux. Even more confusingly, i686 and x86_64 musl Linux builds succeeded.

No worry about this one. `ieee_arithmetic` has been removed. It should not occur if the compilation is done again. Thanks.

**Comment by amontoison** [comment]
@zaikunzhang 
Should we remove the Windows platforms for now?
https://github.com/libprima/prima/issues/58

**Comment by zaikunzhang** [comment]
> @zaikunzhang Should we remove the Windows platforms for now? [libprima/prima#58](https://github.com/libprima/prima/issues/58)

Yes, if this is the only way at the moment.

**Comment by zaikunzhang** [comment]
Hi @amontoison , Windows is supported now. Could you update the recipe accordingly? Thanks.

**Comment by amontoison** [comment]
Done in #7343

**Comment by zaikunzhang** [comment]
Merci!

**Comment by emmt** [comment]
In the `Project.toml` file of `PRIMA_jll`, version is `v0.7.0+0` or  `v0.7.0+1` but this cannot be specified in the `compat` section of my own `Project.toml` file:

```toml
[compat]
julia = "1.6"
PRIMA_jll = "0.7.0+1"
```

yields the following error `invalid version specifier: "0.7.0+1"` in `Pkg.instantiate()`. 

Specifying `0.7.0` for the `compat` section works but since the API of the C functions have chaged between 0.7.0+0 and 0.7.0+1, I really want to distinguish the two versions.

Maybe I am not correctly specifying the version. Otherwise, I think that at least the minor version number should be incremented when the API changes. 

Since the C API of the PRIMA library has changed again since then, it may be a good idea to rebuild the artifact (with the new library and suitable version number).

**Comment by amontoison** [comment]
Hi @emmt!
You just need to provide `PRIMA_jll = "0.7"`.

**Comment by amontoison** [comment]
It's not relevant to distinguish the two versions because the version 0.7.0+0 is replaced by 0.7.0+1. The version 0.7.0+1 will be automatically download.
The number `+i` is a just a way to recompile a package without bumping the version number of the artifact.

**Comment by amontoison** [comment]
> Since the C API of the PRIMA library has changed again since then, it may be a good idea to rebuild the artifact (with the new library and suitable version number).

The C API of the PRIMA library didn't changed since the version 0.7.
Did I miss something?

**Comment by emmt** [comment]
@amontoison thank you for the help. You may be intersted in [PRIMA.jl](https://github.com/emmt/PRIMA.jl) a first attempt I have made to have a Julia interface to the algorithms of `PRIMA` through the artifact `PRIMA_jll`. I did search for such an interface but found none.

**Comment by emmt** [comment]
> The C API of the PRIMA library didn't changed since the version 0.7.
> Did I miss something?

The C API did change: parameters `npt` and `ftarget` were added to `bobyqa`, `newuoa`, and `lincoa` and `ftarget` was added to `cobyla`.

**Comment by zaikunzhang** [comment]
> The C API did change: parameters npt and ftarget were added to bobyqa, newuoa, and lincoa and ftarget was added to cobyla.

Yes. We decided to change a bit the API. This should not happen so often after some iterations. They will be announced if they do happen.

**Comment by zaikunzhang** [comment]
> Yes. We decided to change a bit the API. This should not happen so often after some iterations. They will be announced if they do happen.

Indeed, the Fortran API has been fixed. It was the C API that changed. The C API has not existed long yet.

**Pull request merged**